src/compiler/transformers/classFields.ts

```diff
@@ -937,6 +1064,12 @@ namespace ts {
                                 privateIdentifierInfo.weakMapName
                             );
                         }
+                        case PrivateIdentifierPlacement.StaticField: {
+                            return createPrivateStaticFieldInitializer(
+                                privateIdentifierInfo.variableName,
+                                visitNode(property.initializer, visitor, isExpression)

```

Comment:
What if the initializer uses something like `this.#myFoo`? I think `this` is bound as the class when the initializer is being executed.

Reply:
@joeywatts using `this` in static initializers is forbidden by TS at the moment. There is another ticket that is winding it's way to solve this. I wouldn't add any extra scope to this PR to attempt to fix this for private fields only.